### PR TITLE
✨ [Add] Unit tests for Extension reconciler

### PR DESCRIPTION
**Pull request by varshaprasad96**
# Description
This PR:
- Removes the additional `HasKappAPI` test, since the feature is anyway protected using featureGate. This also helps in running tests easily.
- Adds unit tests for Extension controller. 

## Reviewer Checklist

- [ ] API Go Documentation
- [ ] Tests: Unit Tests (and E2E Tests, if appropriate)
- [ ] Comprehensive Commit Messages
- [ ] Links to related GitHub Issue(s)

### CODE REVIEW

**Review by joelanford — COMMENTED** [review]
It seems like there may be an opportunity to collapse some of the test code because it looks like a decent chunk is repetitive between different tests.

You might consider a table-driven set of tests, where a test case could contain arbitrary setup and assertion functions that could be implemented per test case.

File: internal/controllers/extension_controller_test.go [hunk 1]
```diff
@@ -92,6 +86,931 @@ func TestExtensionReconcile(t *testing.T) {
 	}
 }
 
+func TestExtensionNonExistentPackage(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, features.OperatorControllerFeatureGate, features.EnableExtensionAPI, true)()
```
Comment (dtfranz) [inline internal/controllers/extension_controller_test.go:90]: These new tests looked like they had similar enough structure to potentially be done in [this kind of format](https://github.com/operator-framework/operator-controller/blob/38da6fc3c70c60d4da570333a3e5fc005192d44f/internal/catalogmetadata/types_test.go#L18). Seems to me like that might be more maintainable?

Comment (dtfranz) [reply internal/controllers/extension_controller_test.go:90]: Oops, just realized this is a duplicate comment of Joe's [here](https://github.com/operator-framework/operator-controller/pull/690#pullrequestreview-1931627893).

**Comment by perdasilva** [comment]
closing due to inactivity - please re-open if we still need this =D

**Pull request closed without merging**